jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -335,6 +335,13 @@ public void onUpgradeTo(ByteBuffer buffer)
         BufferUtil.append(getRequestBuffer(), buffer);
     }
 
+    @Override
+    public void upgrade()
+    {
+        getParser().upgrade();
+        getGenerator().upgrade();
+    }
```

Comment:
Sorry, but I've changed my mind about the naming here.   This is not really `upgrade`, it would be more accurate to call these methods `tunnel` or `openTunnel` or `startTunnel`.

The description would be to convert the connection over which the current protocol us running to a tunnel, but without replacing the connection (as is done by our real upgrade mechanism).  This can be used for upgrade within a connection, but it is not really an upgrade for this connection, as the connection remains and just tunnels data to/from its endpoint.